Misc/NEWS.d/next/Library/2021-01-19-10-49-55.bpo-42896.emKDJ-.rst

```diff
@@ -0,0 +1 @@
+Allow for Solaris 11.4 crle output not containing ELF
```

Comment:
Please use full sentences in news entries.  Also, this entry is too vague -- please add more details.

Reply:
Will try to update the PR title once but would have thought while terse, it stated the issue well enough, and take a look at unit tests for this ( simple enough to test, just another test framework to learn :) )

Reply:
PR title updated

Reply:
Looking at existing tests at the end of util.py and ctypes/test

util.py  def test
```
344     if os.name == "posix":
345         # find and load_version
346         print(find_library("m"))
```

will fail and print nothing on Solaris 11.4, for example:
without the fix
```
root@sol114:/export/home/david# python3
Python 3.7.8 (default, Jan 14 2021, 23:45:01) 
[GCC 7.3.0] on sunos5
Type "help", "copyright", "credits" or "license" for more information.
>>> import ctypes.util
>>> ctypes.util.find_library('m')
>>> 
```

with the fix
```
root@sol114:/export/home/david# python3
Python 3.7.8 (default, Jan 14 2021, 23:45:01) 
[GCC 7.3.0] on sunos5
Type "help", "copyright", "credits" or "license" for more information.
>>> import ctypes.util
>>> ctypes.util.find_library('m')
'libm.so.2'
>>> 
```

Similarly the same occurs finding the library 'c'. Hence I believe that the setupModule of ctypes/test/test_loading.py catches this case but I am perplexed by class LoaderTest.test_load where if libc is None, skiptest ??.  Surely this should be an error if libc cannot be found on a system then something is wrong.  Similarly LoaderTest.test_find, libraries 'c' and 'm' would not be found on Solaris 11.4 but no action.
Maybe there are some platforms where Python is used and no libc would be available by default.

I include the lines as example https://github.com/python/cpython/blob/b798ab06937f8bb24b444a49dd42e11fff15e654/Lib/ctypes/test/test_loading.py#L10-L53

Existing tests are already catching the condition but are effectively ignored, hence unsure what my test cases would contribute to this when existing tests are ignoring the condition, especially for such a simple fix.

Reply:
@ZackerySpytz wondering about a reply concerning my point about unit tests for this.